docs/sources/mimir/query/query-best-practices.md

```diff
@@ -0,0 +1,84 @@
+- Alerting rules: Alerting rules should almost always operate on short, recent time windows to detect current states or recent trends, rather than long-term historical patterns. Timely alerts depend on the quick evaluations of fresh data.
+- Ad-hoc analysis: When exploring data, start with small time ranges and gradually expand if more historical context is required for your investigation. This iterative approach prevents unnecessary strain on the system during exploratory phases.
+
+If you don't specify a time range in your query, consider using the `-store.max-labels-query-length` CLI flag to limit the maximum time range.
+
+Additionally, choose a step interval that matches your requirements for running the query. Smaller step intervals increase a query's cost but provide higher resolution.
+
+## Match your query time range to the evaluation frequency
+
+Align the time range of your query with how frequently the query is evaluated or how current the data needs to be. For instance, if a query runs every minute, such as for an alerting rule or a frequently refreshing dashboard panel, there's no need for it to look back for a long time period, such as 30 days.
+
+Queries with very long lookback periods can be computationally expensive. They require Mimir to process a large volume of historical data, even if only the latest value is of interest.
+
+## Filter early in the query
+
+Avoid running broad, large-scale queries and then filtering the results within Grafana. This approach is computationally expensive and inefficient. Instead, filter your data as early as possible in the query itself.
+
+Filtering data in the query provides the following benefits:
+
+- Reduced data transfer: Filtering at the query level means Mimir sends only the data you need to Grafana, which reduces network overhead.
+- Lower processing load: Mimir can apply filters at the storage layer, allowing it to discard irrelevant time series and data points before they are retrieved, processed, and aggregated. This significantly reduces the computational load on the Mimir cluster.
+- Improved query speed: Queries that are pre-filtered are generally faster to run and return results.
+
+Use these strategies to implement early filtering:
+
+- Use label selectors: Leverage precise label selectors within your PromQL query to narrow down the dataset from the start. For more information, refer to [Use Precise Label Selectors](#use-precise-label-selectors).
+- Apply functions and aggregations: If you need to transform or aggregate data, do so within the query. For example, `sum by (job) (metric_name)` is more efficient than retrieving all `metric_name` series and then summing them in Grafana.
+
+## Avoid high cardinality operations
+
+Avoid high cardinality operations to avoid performance issues or latency with your queries. High cardinality operations typically involve processing exceptionally large datasets.
```

Comment:
This seems like a duplicate of the "filter early in the query" and "use precise label selectors" sections.